src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -159,15 +159,16 @@ public static string Concat<T>(IEnumerable<T> values)
 
                     // Create the StringBuilder, add the chars we've already enumerated,
                     // add the rest, and then get the resulting string.
-                    StringBuilder result = StringBuilderCache.Acquire();
+                    Span<char> initialBuffer = stackalloc char[64];
```

Comment:
The stack reserved sizes in this file can be probably be a bit higher, like 128 chars.

Reply:
We've been fairly loose / non-prescriptive with how we do stackallocs like this in the past.  In various places around corelib, we use various limits for stackalloc char when we don't know the actual upper limit, including 256, 128, 64, and 32 (lots of other sizes are used of course when we do have an upper bound).  Not as part of this change, but it seems we should probably standardize with some const somewhere that everyone just uses.

Reply:
I agree that a general reasonable sized buffer constant would be useful.

We often know the typical maximum size, e.g. 32-chars for a number formatting. I assume that we would keep these scenario specific maximum sizes.

Reply:
> I assume that we would keep these scenario specific maximum sizes.

Yup.

Reply:
Might also want to harmonize with the jit's threshold for converting fixed sized locallocs to locals, which is currently 32. [Back when I set this](https://github.com/dotnet/coreclr/pull/14623#issue-147876729) I did not see many > 32 cases.

Reply:
Here we have `char`, so that limit would be 16 (= 32 bytes)?